adafruit_gps.py

```diff
@@ -101,11 +102,13 @@ def update(self):
         sentence = self._parse_sentence()
         if sentence is None:
             return False
+        if self.debug:
+            print(sentence)
         data_type, args = sentence
-        data_type = data_type.upper()
-        if data_type == 'GPGGA':      # GGA, 3d location fix
+        data_type = bytes(data_type.upper(), "utf-8")
```

Comment:
How about using `"ascii"` here and below as the encoding? In CircuitPython this arg is ignored (you can say `"foobar"`), but in regular Python it has to be an encoding name. If pyserial is reading junk with chars >128, then it will throw an error:
```
>>> bytes('ab\xbb', 'ascii')
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
UnicodeEncodeError: 'ascii' codec can't encode character '\xbb' in position 2: ordinal not in range(128)
>>> bytes('ab\xbb', 'utf-8')
b'ab\xc2\xbb'
>>> 
```